python/paddle/fluid/parallel_executor.py

```diff
@@ -95,7 +95,14 @@ def __init__(self,
         self._places = []
         self._act_places = []
         if use_cuda:
-            for i in six.moves.range(core.get_cuda_device_count()):
+            gpus = []
+            gpus_env = os.getenv("PADDLE_GPUS")
```

Comment:
Can we do that using 
```
# gpu ids or cpu ids. The ids are mapped to the underlying gpus or cpus and each id corresponds to a model run in data-parallel.
build_strategy.devices
```
And deprecate CPU_NUM and PADDLE_GPUS?

Reply:
I'm thinking of this too, currently, the `init_devices` is called when importing paddle modules, since all those places should be configured with the GPU list, we have to use this for now. Setting build_strategy only affects ParallelExecutor. Will do deprecate `CPU_NUM` and `PADDLE_GPUS` in later on PRs.